### PR TITLE
Child chain mintable predicates E2E tests

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

This PR tweaks CLI commands for sending bridge transactions and implements E2E tests which prove that child chain mintable predicates are working correctly.
Note that child chain mintable native token is mapped only on mintable predicates (it is prevented to be mapped in non-mintable ones).

## Bridge CLI commands UX changes
Bridge deposit commands have a new flag called `--minter-key`, which represents the private key of the account that is able to do token minting (e.g. has associated `MINTER_ROLE` etc.). If it is provided, deposit command is going to mint tokens to the sender account (otherwise it would do just deposit).

All bridge commands are enhanced with an additional flag called `--child-chain-mintable`. This flag denotes whether child chain mintable predicates are engaged in a given bridge transaction. It is used to determine whether exit events (aka `L2StateSynced` events) are extracted from a receipt. Depending on the direction of assets bridging, exit events can be present either:
- for withdrawals when using non-child chain mintable predicates
- for deposits when using child chain mintable predicates

## Mandatory `stake-token` flag to `rootchain deploy` command
Introduced mandatory stake-token flag to `rootchain deploy` command, because the previous approach was not correct, since instead of using root chain native token, we were using child chain native token root chain pairing address.

## E2E tests
Implemented two e2e tests, which are bridging child chain-originated tokens (mintable native tokens and ERC721 tokens).

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: command/bridge/withdraw/erc721/withdraw_erc721.go [hunk 1]
```diff
@@ -191,44 +173,29 @@ func createWithdrawTxn(receivers []ethgo.Address, tokenIDs []*big.Int) (*ethgo.T
...
 	var buffer bytes.Buffer
 
 	vals := make([]string, 0, 5)
 	vals = append(vals, fmt.Sprintf("Sender|%s", r.Sender))
 	vals = append(vals, fmt.Sprintf("Receivers|%s", strings.Join(r.Receivers, ", ")))
-	vals = append(vals, fmt.Sprintf("TokenIDs|%s", strings.Join(r.TokenIDs, ", ")))
-	vals = append(vals, fmt.Sprintf("Exit Event IDs|%s", r.ExitEventID))
+	vals = append(vals, fmt.Sprintf("Token IDs|%s", strings.Join(r.TokenIDs, ", ")))
+
+	if wp.ChildChainMintable {
```
Comment (igorcrevar) [inline command/bridge/withdraw/erc721/withdraw_erc721.go]: can we change this to `if r.ExitEventId != ""`

File: command/bridge/withdraw/erc1155/withdraw_erc1155.go [hunk 2]
```diff
@@ -246,7 +211,11 @@ func (r *withdrawResult) GetOutput() string {
 	vals = append(vals, fmt.Sprintf("Receivers|%s", strings.Join(r.Receivers, ", ")))
 	vals = append(vals, fmt.Sprintf("Amounts|%s", strings.Join(r.Amounts, ", ")))
 	vals = append(vals, fmt.Sprintf("Token IDs|%s", strings.Join(r.TokenIDs, ", ")))
-	vals = append(vals, fmt.Sprintf("Exit Event ID|%s", r.ExitEventID))
+
+	if !wp.ChildChainMintable {
```
Comment (igorcrevar) [inline command/bridge/withdraw/erc1155/withdraw_erc1155.go]: `if r.ExitEventId != ""`

File: command/bridge/deposit/erc721/deposit_erc721.go [hunk 3]
```diff
@@ -272,21 +270,26 @@ func createApproveERC721PredicateTxn(rootERC721Predicate, rootERC721Token types.
...
+func (r *depositResult) GetOutput() string {
 	var buffer bytes.Buffer
 
 	vals := make([]string, 0, 3)
 	vals = append(vals, fmt.Sprintf("Sender|%s", r.Sender))
 	vals = append(vals, fmt.Sprintf("Receivers|%s", strings.Join(r.Receivers, ", ")))
-	vals = append(vals, fmt.Sprintf("TokenIDs|%s", strings.Join(r.TokenIDs, ", ")))
+	vals = append(vals, fmt.Sprintf("Token IDs|%s", strings.Join(r.TokenIDs, ", ")))
+
+	if dp.ChildChainMintable {
```
Comment (igorcrevar) [inline command/bridge/deposit/erc721/deposit_erc721.go]: `if r.ExitEventID != ""`

File: command/bridge/deposit/erc20/deposit_erc20.go [hunk 4]
```diff
@@ -254,21 +248,26 @@ func createDepositTxn(sender, receiver types.Address, amount *big.Int) (*ethgo.T
...
+func (r *depositResult) GetOutput() string {
 	var buffer bytes.Buffer
 
 	vals := make([]string, 0, 3)
 	vals = append(vals, fmt.Sprintf("Sender|%s", r.Sender))
 	vals = append(vals, fmt.Sprintf("Receivers|%s", strings.Join(r.Receivers, ", ")))
 	vals = append(vals, fmt.Sprintf("Amounts|%s", strings.Join(r.Amounts, ", ")))
 
-	buffer.WriteString("\n[DEPOSIT ERC 20]\n")
+	if dp.ChildChainMintable {
```
Comment (igorcrevar) [inline command/bridge/deposit/erc20/deposit_erc20.go]: `if r.ExitEventID != ""`

File: command/bridge/deposit/erc20/deposit_erc20.go [hunk 5]
```diff
@@ -214,21 +198,31 @@ func runCommand(cmd *cobra.Command, _ []string) {
 					return fmt.Errorf("receiver: %s, amount: %s", receiver, amount)
 				}
 
+				if dp.ChildChainMintable {
+					exitEventID, err := common.ExtractExitEventID(receipt)
+					if err != nil {
+						return fmt.Errorf("failed to extract exit event: %w", err)
+					}
+
+					exitEventIDs[i] = strconv.FormatUint(exitEventID.Uint64(), 10)
```
Comment (igorcrevar) [inline command/bridge/deposit/erc20/deposit_erc20.go]: maybe we should keep events as []*big.Int and then in actual output convert from *big.Int to string

File: command/bridge/deposit/erc1155/deposit_erc1155.go [hunk 6]
```diff
@@ -320,6 +319,10 @@ func (r *depositERC1155Result) GetOutput() string {
 	vals = append(vals, fmt.Sprintf("Amounts|%s", strings.Join(r.Amounts, ", ")))
 	vals = append(vals, fmt.Sprintf("Token IDs|%s", strings.Join(r.TokenIDs, ", ")))
 
+	if dp.ChildChainMintable {
```
Comment (igorcrevar) [inline command/bridge/deposit/erc1155/deposit_erc1155.go]: `if r.ExitEventID != ""`

File: command/bridge/deposit/erc1155/deposit_erc1155.go [hunk 7]
```diff
@@ -227,13 +213,25 @@ func runCommand(cmd *cobra.Command, _ []string) {
...
+
+	if dp.ChildChainMintable {
+		exitEventID, err := common.ExtractExitEventID(receipt)
+		if err != nil {
+			outputter.SetError(fmt.Errorf("failed to extract exit event: %w", err))
+
+			return
+		}
+
+		res.ExitEventID = strconv.FormatUint(exitEventID.Uint64(), 10)
```
Comment (igorcrevar) [inline command/bridge/deposit/erc1155/deposit_erc1155.go]: Maybe we can keep this data as *big.Int and then outputter should convert it to string

File: command/bridge/withdraw/erc1155/withdraw_erc1155.go (hunk 2 above)
Comment (Stefan-Ethernal) [reply command/bridge/withdraw/erc1155/withdraw_erc1155.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1560/commits/a97504b37b70933f07ebaf37b8a51c26685cb7fc

File: command/bridge/withdraw/erc721/withdraw_erc721.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply command/bridge/withdraw/erc721/withdraw_erc721.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1560/commits/a97504b37b70933f07ebaf37b8a51c26685cb7fc

File: command/bridge/deposit/erc721/deposit_erc721.go (hunk 3 above)
Comment (Stefan-Ethernal) [reply command/bridge/deposit/erc721/deposit_erc721.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1560/commits/a97504b37b70933f07ebaf37b8a51c26685cb7fc

File: command/bridge/deposit/erc20/deposit_erc20.go (hunk 4 above)
Comment (Stefan-Ethernal) [reply command/bridge/deposit/erc20/deposit_erc20.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1560/commits/a97504b37b70933f07ebaf37b8a51c26685cb7fc

File: command/bridge/deposit/erc1155/deposit_erc1155.go (hunk 6 above)
Comment (Stefan-Ethernal) [reply command/bridge/deposit/erc1155/deposit_erc1155.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1560/commits/a97504b37b70933f07ebaf37b8a51c26685cb7fc

File: command/bridge/deposit/erc1155/deposit_erc1155.go (hunk 7 above)
Comment (Stefan-Ethernal) [reply command/bridge/deposit/erc1155/deposit_erc1155.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1560/commits/2a3d80c845b44ca5393a65c6bc78df229ff1e31d

File: command/bridge/deposit/erc20/deposit_erc20.go (hunk 5 above)
Comment (Stefan-Ethernal) [reply command/bridge/deposit/erc20/deposit_erc20.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1560/commits/2a3d80c845b44ca5393a65c6bc78df229ff1e31d

**Review by Nemanja0x — APPROVED** [review]

**Review by igorcrevar — APPROVED** [review]

**Pull request merged**